src/test/kotlin/nl/info/zac/authentication/UserPrincipalFilterTest.kt

```diff
@@ -116,8 +120,12 @@ class UserPrincipalFilterTest : BehaviorSpec({
         every { accessToken.name } returns "fakeFullName"
         every { accessToken.email } returns "fakeemail@example.com"
         every { accessToken.getStringListClaimValue("group_membership") } returns emptyList()
-        every { zaakafhandelParameterService.listZaakafhandelParameters() } returns zaakafhandelParameters
         every { newHttpSession.setAttribute(any(), any()) } just runs
+        every {
+            pabcClientService.getApplicationRoles(
+                match { it == expectedFunctionalRoles }
```

Comment:
I have never seen this `match` construct. It that really needed? Can you just do `pabcClientService.getApplicationRoles(expectedFunctionalRoles)` here?